doc/pipeline_details.rst

```diff
@@ -0,0 +1,31 @@
+.. _pipeline-details:
+
+Pipeline Details
+================
+
+Ibex has a 2 stage pipeline, the 2 stages are:
+
+Instruction Fetch (IF)
+  Fetches instructions from memory via a prefetch buffer, capable of fetching 1 instruction per cycle if the instruction side memory system allows. See :ref:`instruction-fetch` for details.
+
+Instruction Decode and Execute (ID/EX)
+  Decodes fetched instruction and immediately executes it, register read and write all occurs in this stage. Multi-cycle instructions will stall this stage until they are complete See :ref:`instruction-decode-execute` for details.
+
+All instructions required two cycles minimum to pass down the pipeline. One cycle in the IF stage and one in the ID/EX stage. Not all instructions can complete in the ID/EX stage in one cycle so will stall there until they complete. This means the maximum CPI (clocks per instruction) Ibex can achieve is 1 when multi-cycle instructions aren't used. See Multi and Single Cycle Instructions below for the details.
+
+Multi and Single Cycle Instructions
+-----------------------------------
+
+All of the 'Integer Computational Instructions' as defined in the RISC-V RV32I Base Integer Instruction set are single cycle. The CSR access instructions defined in 'Zicsr' of the RISC-V specification are also single cycle. All other instructions are multi-cycle. Below are descriptions of the multi-cycle operations for various categories of instructions.
```

Comment:
Base Integer Instruction Set

Reply:
"multi cycle" but "multi-cycle instructions"

Reply:
I think the rendering of sub-headings below is quite nice and it might be worth considering re-arranging to just list instruction categories and indicate single/multi-cycle below each heading. I don't feel strongly about this, just throwing out the idea.

e.g.

## RV32I Integer Computational Instructions
*Single cycle*. [description]

## Zicsr CSR access instructions
*Single cycle*. [description]

## Load/Store operations
*Multi cycle*. [description]

etc

Reply:
I was wondering if I should add a table of instruction latencies.  Perhaps I could put the descriptions in that same table but maybe it's better to have a smaller easily readable table for people who just want the number and don't care about the reason then what @asb suggests for the long-form description?

Reply:
Nice table!